### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Write proper integration tests for the MicroK8s charm, using the pytest-operator.

Adds configuration options for the version matrix to test (charm version X MicroK8s snap version).

Also applies minor fixes in the charm code to prevent interminent issues from crashing the charm (and so the integration tests).

Also, rewrite the GitHub actions that tests building the charm to run our integration tests.

### Notes

This is a pre-requisite for taking ownership of the MicroK8s charm.

### CODE REVIEW

**Comment by neoaggelos** [comment]
@ktsakalozos

File: README.md [hunk 1]
```diff
@@ -10,12 +10,43 @@ to the resources required by the applications you plan to deploy.
 
 ## Usage
 
-    juju deploy --constraints 'cores=2 mem=4G' cs:~pjdc/microk8s
+```bash
+juju deploy --constraints 'cores=2 mem=4G' cs:~pjdc/microk8s
```
Comment (ktsakalozos) [inline README.md:14]: Is the pjdc the right namespace?

Comment (neoaggelos) [reply README.md]: Updated to new CharmHub name

**Review by ktsakalozos — APPROVED** [review]
LGTM +1 thank you @neoaggelos

**Pull request merged**